### PR TITLE
add exists alias

### LINKED ISSUE #1225: expect(...).to.have.property(...).that.exists
`expect(...).to.have.property(...).that.exists`;

Simply adding **exists** as an alias for `exist` would support such idiomatic cases. If there is support for this I can create a PR.

### PR DESCRIPTION
**Pull request by blake-regalia**
Closes #1225 .

### CODE REVIEW

File: lib/chai/core/assertions.js [hunk 1]
```diff
@@ -814,23 +814,32 @@ module.exports = function (chai, _) {
    *     expect(undefined).to.be.undefined; // Recommended
    *     expect(undefined).to.not.exist; // Not recommended
    *
+   * Use after `.property` to assert that a value is not `null` nor `undefined`.
+   *     expect({a: 1}}).to.have.property('a').that.exists;
```
Comment (meeber) [inline lib/chai/core/assertions.js]: I'm not sure I like this example. For the most part, we discourage the use of `.exists` in favor of a precise assertion whenever reasonable. For example:

```js
expect({a: 1}).to.have.property('a', 1); // Recommended
expect({a: 1}).to.have.property('a').that.equals(1); // Recommended
expect({a: 1}).to.have.property('a').that.exists; // Not recommended
```

File: test/should.js [hunk 2]
```diff
@@ -257,7 +257,9 @@ describe('should', function() {
     var foo = 'foo'
       , bar = undefined;
     should.exist(foo);
+    should.exists(foo);
     should.not.exist(bar);
+    should.not.exists(bar);
```
[the remark this answers is not in the document]
Comment (meeber) [reply test/should.js]: @blake-regalia Do you have a few min to remove these `.should` tests? Apologies for the inconvenience.

[the remark this answers is not in the document]
Comment (blake-regalia) [reply test/should.js]: No problem! Thanks for following up with this.

**Comment by vieiralucas** [comment]
This PR looks good, but I'm not sure if we should merge this since we will start merging `new-chai` into this repo.
Should we wait for after `new-chai` is here to add this assertion? Will `chai 5` even have an `exists` assertion?

@keithamus ?

**Comment by lucasfcosta** [comment]
@vieiralucas I'm happy to get this merged in as it's an improvement to the current codebase that has 0 cost.
IMO our approach to v4 is to get contributions and improvements that don't add any overhead or do critical fixes but avoid putting too much effort into it as we move towards v5 since that's our main goal atm.

**Comment by vieiralucas** [comment]
Yes, I got it wrong about `new-chai`
But I believe we should merge #1249 before merging this since it should be easier to fix conflicts.

**Review by keithamus — APPROVED** [review]
Excellent, let's get this in!

**Pull request merged**